### DISCUSSION TITLE
v1.61.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#2780] [Announcements]
# v1.61.0

Welcome to the `v1.61.0`  release of Immich. There are many updates in this version that we hope you'll like. Some of the key highlights include:

* Client-side hashing on the mobile app
* Justified layout for timeline view on the web
* Proper support for RAW file formats
* Memories feature on the web
* Select-all button on all web views

### Client-side hashing on the mobile app

The mobile app will now calculate the hash of each asset when an album(s) are selected for backup from the backup screen. 

This mechanism will provide a way to implement additional features in the future to improve the deduplication mechanism on the mobile app. So in the event of changing the mobile device and copying the existing gallery from the older device to the newer one, the mobile client will be able to identify which assets are on the server to avoid re-uploading them.

After the hash is calculated, the local asset will show up on the timeline.

When viewing the Photos page, you can check the hashing progress via the loading indicator on the bottom app bar.

![image](https://github.com/immich-app/immich/assets/27055614/a25a08c0-e1b9-4dc0-9df6-19bdc660527c)

The hashing speed is optimized for Android (using native Kotlin) but not yet for iOS devices.

### Justified timeline layout

This feature considers the orientation of the photos and videos to display them with the correct aspect ratio in a justified layout manner. These features improve the viewing experiences with an aesthetic feeling on the main timeline.

![image](https://github.com/immich-app/immich/assets/27055614/7017c795-59e5-4f63-85fd-7cf572e2e137)

### Proper support of RAW file formats

Thanks to our new contributor @uhthomas, we are now using a better library to generate thumbnails for RAW files, specifically Sony ARW in this version, so that they are no longer blurry. Please consider rerunning the thumbnail generation job if you uploaded RAW files before this release.

### Memories feature

This release adds a foundation for the memories feature you love in other gallery applications. The first iteration implements the mechanism to remind you of events that happened x years ago and go back 30 years to the past. The memories are available to view as a gallery of the whole day or a slideshow with the ability to autoplay.

https://github.com/immich-app/immich/assets/27055614/01cd3dbb-d30f-4022-a58f-2289daf00a88

### Select-all button on the web

You can now select all assets from the timeline, album, and other views on the web

![image](https://github.com/immich-app/immich/assets/27055614/00395008-9b26-4765-be3e-3fd9ed80c174)

----
And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="Loading ~4000 images/videos"> 
</p>

If you find the project helpful and helps you in some ways, you can support it via the following channel.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

🎉 Cheers! 🎉

### Server
* refactor(server): move asset checks to service by @jrasm91 in https://github.com/immich-app/immich/pull/2640
* refactor(server): partner core by @jrasm91 in https://github.com/immich-app/immich/pull/2678
* refactor(server): album download check by @jrasm91 in https://github.com/immich-app/immich/pull/2679
* refactor(server): shared link asset access check by @jrasm91 in https://github.com/immich-app/immich/pull/2680
* refactor(server,web): add/remove album users by @jrasm91 in https://github.com/immich-app/immich/pull/2681
* refactor(server): bootstrap code by @jrasm91 in https://github.com/immich-app/immich/pull/2682
* refactor(server)*: tsconfigs by @jrasm91 in https://github.com/immich-app/immich/pull/2689
* fix(server): Filter out deleted partners by @brighteyed in https://github.com/immich-app/immich/pull/2697
* chore(server): fix test coverage by @jrasm91 in https://github.com/immich-app/immich/pull/2699
* chore(server): remove unused code by @jrasm91 in https://github.com/immich-app/immich/pull/2700
* chore(server): sort open api spec keys by @jrasm91 in https://github.com/immich-app/immich/pull/2710
* feat(server): option to transcode to original resolution by @mertalev in https://github.com/immich-app/immich/pull/2709
* chore(server): update node images by @uhthomas in https://github.com/immich-app/immich/pull/2736
* feat(server): use imagemagick and libraw for raw image support by @uhthomas in https://github.com/immich-app/immich/pull/2668
* chore(server): don't fallback to ExifTool for embedded image previews by @uhthomas in https://github.com/immich-app/immich/pull/2747
* feat(server): allow emails without a tld by @uhthomas in https://github.com/immich-app/immich/pull/2762
* fix(server): increase request timeout from 5 minutes to 30 minutes by @zackpollard in https://github.com/immich-app/immich/pull/2769

### Web
* feat(web, server): Implement justified layout for AssetGrid by @brighteyed in https://github.com/immich-app/immich/pull/2666
* feat(web): mobile job cards by @jrasm91 in https://github.com/immich-app/immich/pull/2688
* fix(web): Timeline narrow date groups style by @brighteyed in https://github.com/immich-app/immich/pull/2713
* feat(web): Add select all button to all views by @Y0ngg4n in https://github.com/immich-app/immich/pull/2714
* feat(web): support uploading Insta360 file format by @truongsinh in https://github.com/immich-app/immich/pull/2725
* fix(web): empty user initials by @jrasm91 in https://github.com/immich-app/immich/pull/2737
* feat(web): Memory by @alextran1502 in https://github.com/immich-app/immich/pull/2759

### Mobile
* feature(mobile): hash assets & sync via checksum by @fyfrey in https://github.com/immich-app/immich/pull/2592
* fix(mobile): disable hit testing for transparent bars by @brighteyed in https://github.com/immich-app/immich/pull/2727
* feat(mobile): sync remote assets without thumbs by @fyfrey in https://github.com/immich-app/immich/pull/2705

### Machine Learning
* chore(ml): updated dockerfile, added typing, packaging by @mertalev in https://github.com/immich-app/immich/pull/2642
* feat(ml): model unloading by @mertalev in https://github.com/immich-app/immich/pull/2661

### Others
* docs: Update backup/restore example to address SQL errors on database import. by @wittymap in https://github.com/immich-app/immich/pull/2663
* chore(deps): bump docker/setup-qemu-action from 2.1.0 to 2.2.0 by @dependabot in https://github.com/immich-app/immich/pull/2695
* chore(deps): bump docker/setup-buildx-action from 2.5.0 to 2.6.0 by @dependabot in https://github.com/immich-app/immich/pull/2696
* docs: fix Immich internal address while bulk uploading through docker by @evanphilip in https://github.com/immich-app/immich/pull/2704
* chore(deps): bump docker/build-push-action from 4.0.0 to 4.1.0 by @dependabot in https://github.com/immich-app/immich/pull/2702
* chore(deps): bump docker/setup-buildx-action from 2.6.0 to 2.7.0 by @dependabot in https://github.com/immich-app/immich/pull/2755
* chore(deps): bump docker/build-push-action from 4.1.0 to 4.1.1 by @dependabot in https://github.com/immich-app/immich/pull/2761

## New Contributors
* @wittymap made their first contribution in https://github.com/immich-app/immich/pull/2663
* @evanphilip made their first contribution in https://github.com/immich-app/immich/pull/2704
* @Y0ngg4n made their first contribution in https://github.com/immich-app/immich/pull/2714

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.60.0...v1.61.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.61.0'>v1.61.0</a>.</em>

**Comment by dorianim** [comment]
The work you are doing here is awesome! Thank you so much!

**Comment by alexonpeace** [comment]
Finally, been waiting for the Justified web timeline I can finally switch from gp 
Thanks a lot for your hard work 😉

**Comment by brianread108** [comment]
Can someone give me an "easy" way to update an existing system from 1.60 -> 1.61? I installed initially using the (experimental) install script, but I tried re-using it and the resultant system would not display the web page.

**Reply by bo0tzz** [reply]
https://immich.app/docs/install/docker-compose#step-4---upgrading

**Reply by brianread108** [reply]
Just what I needed - many thanks.
